tests/cel/clientsettingspolicy_test.go

```diff
@@ -0,0 +1,224 @@
+					Kind:  GatewayKind,
+					Group: GatewayGroup,
+				},
+			},
+		},
+		{
+			name:       "Validate invalid.networking.k8s.io TargetRef Group is not allowed",
+			wantErrors: []string{ExpectedTargetRefGroupError},
+			policySpec: ngfAPIv1alpha1.ClientSettingsPolicySpec{
+				TargetRef: gatewayv1alpha2.LocalPolicyTargetReference{
+					Kind:  GatewayKind,
+					Group: InvalidGroup,
+				},
+			},
+		},
+		{
+			name:       "Validate discovery.k8s.io/v1 TargetRef Group is not allowed",
+			wantErrors: []string{ExpectedTargetRefGroupError},
+			policySpec: ngfAPIv1alpha1.ClientSettingsPolicySpec{
+				TargetRef: gatewayv1alpha2.LocalPolicyTargetReference{
+					Kind:  GatewayKind,
+					Group: DiscoveryGroup,
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			validateClientSettingsPolicy(t, tt)
+		})
+	}
+}
+
+func validateClientSettingsPolicy(t *testing.T, tt struct {
+	policySpec ngfAPIv1alpha1.ClientSettingsPolicySpec
+	name       string
+	wantErrors []string
+},
+) {
+	t.Helper()
+	g := NewWithT(t)
+
+	// Get Kubernetes client from test framework
+	// This should be set up by your test framework to connect to a real cluster
+	k8sClient, err := getKubernetesClient(t)
+
+	g.Expect(err).ToNot(HaveOccurred())
+
+	policySpec := tt.policySpec
+	policySpec.TargetRef.Name = gatewayv1alpha2.ObjectName(fmt.Sprintf(TargetRefFormat, time.Now().UnixNano()))
+	policyName := fmt.Sprintf(PolicyNameFormat, time.Now().UnixNano())
+
+	clientSettingsPolicy := &ngfAPIv1alpha1.ClientSettingsPolicy{
+		ObjectMeta: controllerruntime.ObjectMeta{
+			Name:      policyName,
+			Namespace: "default",
+		},
+		Spec: policySpec,
+	}
+
+	err = k8sClient.Create(context.Background(), clientSettingsPolicy)
+
+	// Clean up after test
+	defer func() {
+		_ = k8sClient.Delete(context.Background(), clientSettingsPolicy)
+	}()
+
+	// Check if we expected errors
+	if len(tt.wantErrors) == 0 {
+		if err != nil {
+			g.Expect(err).ToNot(HaveOccurred())
+		}
+		return
+	}
+
+	// We expected errors - validation should have failed
+	if err == nil {
+		g.Expect(err).To(HaveOccurred())
+	}
+
+	// Check that we got the expected error messages
+	for _, wantError := range tt.wantErrors {
+		g.Expect(err.Error()).To(ContainSubstring(wantError), "Expected error '%s' not found in: %s", wantError, err.Error())
+	}
+}
+
+// getKubernetesClient returns a client connected to a real Kubernetes cluster.
+func getKubernetesClient(t *testing.T) (k8sClient client.Client, err error) {
+	t.Helper()
+	// Use controller-runtime to get cluster connection
+	k8sConfig, err := controllerruntime.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	// Set up scheme with NGF types
+	scheme := runtime.NewScheme()
+	if err = ngfAPIv1alpha1.AddToScheme(scheme); err != nil {
+		return nil, err
+	}
+
+	// Create client
+	k8sClient, err = client.New(k8sConfig, client.Options{Scheme: scheme})
+
+	return k8sClient, err
```

Comment:
You could just `return client.New(...)`

Reply:
True. I think I might keep it the way it is if that's okay. I find this to be more readable